File: app/io/pathfinder/config/Global.java
package io.pathfinder.config;

import io.pathfinder.routing.Router;
import play.Application;
import play.GlobalSettings;
import play.Logger;

/**
 * These hooks are called by Play Framework. We can use them to initialize expensive objects and
 * set up any communication channels.
 */
public class Global extends GlobalSettings {
    public void onStart(Application app) {
        Logger.info("Application has started.");
        Router.init();
    }

    public void onStop(Application app) {
        Logger.info("Application has stopped.");
    }
}

File: app/io/pathfinder/config/Global.scala
package io.pathfinder.config

import akka.actor.ActorSystem
import io.pathfinder.routing.Router
import play.api.GlobalSettings
import play.api.Application
import play.Logger
import io.pathfinder.authentication.AuthServer

/**
 * These hooks are called by Play Framework. We can use them to initialize expensive objects and
 * set up any communication channels.
 */
object Global extends GlobalSettings {

    val actorSystem = ActorSystem.create()

    override def onStart(app: Application) {
        Logger.info("Application has started.")
        Router
        AuthServer
    }

    override def onStop(app: Application) {
        Logger.info("Application has stopped.")
    }
}
